node/sc/bridge_manager_test.go

```diff
@@ -1464,16 +1466,130 @@ func TestAnchoringBasic(t *testing.T) {
+
+	bAcc.SetParentOperatorFeePayer(feePayer.Address)
+
+	assert.Equal(t, uint64(0), sc.handler.txCountStartingBlockNumber)
+	assert.Equal(t, uint64(0), sc.handler.latestTxCountAddedBlockNumber)
+	assert.Equal(t, uint64(1), sc.handler.chainTxPeriod)
+
+	// Encoding anchoring tx
+	_, _, _, err = bridge.DeployBridge(tester, sim, true) // dummy tx
+	sim.Commit()
+	curBlk := sim.BlockChain().CurrentBlock()
+
+	// Generate anchoring tx again for only the curBlk.
+	sc.handler.blockAnchoringManager(curBlk)
+	pending := sc.GetBridgeTxPool().Pending()
+	assert.Equal(t, 1, len(pending))
+	var tx *types.Transaction
+	for _, v := range pending {
+		assert.Equal(t, 1, len(v))
+		tx = v[0]
+
+		// Check Balance
+		feePayerBalanceBefore, err := sim.BalanceAt(context.Background(), feePayer.Address, nil)
+		assert.NoError(t, err)
+		parentOperatorBalanceBefore, err := sim.BalanceAt(context.Background(), parentOperator.address, nil)
+		assert.NoError(t, err)
+
+		sim.SendTransaction(context.Background(), tx)
+		sim.Commit()
+
+		// Check Balance
+		feePayerBalanceAfter, err := sim.BalanceAt(context.Background(), feePayer.Address, nil)
+		assert.NoError(t, err)
+		parentOperatorBalanceAfter, err := sim.BalanceAt(context.Background(), parentOperator.address, nil)
+		assert.NoError(t, err)
+
+		receipt, err := sim.TransactionReceipt(context.Background(), tx.Hash())
+		assert.NoError(t, err)
+
+		fee := new(big.Int).SetUint64(receipt.GasUsed * params.DefaultUnitPrice)
+
+		assert.Equal(t, new(big.Int).Sub(feePayerBalanceBefore, fee).String(), feePayerBalanceAfter.String())
+		t.Log("feePayer paid ", fee)
+		assert.Equal(t, parentOperatorBalanceBefore, parentOperatorBalanceAfter)
+	}
+
+	assert.Equal(t, uint64(0), sc.handler.txCount)
+	assert.Equal(t, curBlk.NumberU64(), sc.handler.latestTxCountAddedBlockNumber)
+	compareBlockAndAnchoringTx(t, curBlk, tx)
+}
+
+func generateAnchoringEnv(t *testing.T, tempDir string) (*backends.SimulatedBackend, *SubBridge, *BridgeAccounts, *accountInfo, accounts.Account, *bind.TransactOpts) {
 	config := &SCConfig{AnchoringPeriod: 1}
 	config.DataDir = tempDir
 	config.VTRecovery = true
 
-	bAcc, _ := NewBridgeAccounts(tempDir)
+	ks := keystore.NewKeyStore(tempDir, keystore.StandardScryptN, keystore.StandardScryptP)
+	back := []accounts.Backend{
+		ks,
+	}
+	am := accounts.NewManager(back...)
+	am.Wallets()
+	bAcc, _ := NewBridgeAccounts(am, tempDir, database.NewDBManager(&database.DBConfig{DBType: database.MemoryDB}))
 	bAcc.pAccount.chainID = big.NewInt(0)
 	bAcc.cAccount.chainID = big.NewInt(0)
+	parentOperator := bAcc.pAccount
 
-	alloc := blockchain.GenesisAlloc{}
-	sim := backends.NewSimulatedBackend(alloc)
+	aliceKey, _ := crypto.GenerateKey()
+	alice := bind.NewKeyedTransactor(aliceKey)
+
+	initBal := new(big.Int).Exp(big.NewInt(10), big.NewInt(50), nil)
+
+	feePayer, err := ks.NewAccount("pwd")
+	ks.TimedUnlock(feePayer, "pwd", 0)
+	assert.NoError(t, err)
```

Comment:
How about moviong this line to the L1584?

Reply:
Done. :) Thanks.